[Lee Yi Lin] iP

yleeyilin opened the pull request:
# Duke Ylexi 
> "Your mind is for having ideas, not holding them." - David Allen ([Sources](https://dansilvestre.com/productivity-quotes))
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use 
All you need to do is, 
1. download it from [here](https://github.com/yleeyilin/ip)
2. double-click it
3. add your tasks
4. let it manage your tasks for you😉

And it is **FREE!**

Features: 
- [x] Managing tasks 
- [x] Managing deadlines
- [ ] Reminders (coming soon) 

#
Check out Duke Ylexi! Here's the `main` method: 
```
package duke;

import duke.task.TaskList;

/**
 * Represents a chat bot that helps user keep track of tasks. 
 */
public class Duke {
    private Storage storage;
    private TaskList tasks;
    private Ui ui;
    private Parser parser;

    /**
     * Constructs an instance of the Duke Chatbot.
     * 
     * @param filePath The filepath where task list data is stored. 
     */
    public Duke(String filePath) {
        try {
            this.storage = new Storage(filePath);
            this.tasks = new TaskList(this.storage.loadTaskList()); 
            this.parser = new Parser(); 
        } catch (DukeException e) {
            System.out.println(e.toString());
        }
    }

    /**
     * Runs the program until the "bye" instruction is inputted by the user and interprets the commonad word. 
     */
    public void run() {
        this.ui = new Ui(System.in); 
        System.out.println(this.ui.greet());

        String instr = this.ui.getInstr(); 
        while (!instr.equals("bye")) {
            this.parser.parseInstr(this.tasks, this.storage, instr);
            instr = this.ui.getInstr(); 
        }
        System.out.println(this.ui.exit());
        this.ui.close();
    }

    /**
     * Calls the constructor and is the entry point of the program. 
     * 
     * @param args Input by the user 
     */
    public static void main(String[] args) {
        new Duke("src/main/data/caching.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yleeyilin committed:
greet and exit

yleeyilin committed:
echo

yleeyilin committed:
add and list functions

yleeyilin committed:
mark as done

yleeyilin committed:
todo, event, deadlines

yleeyilin committed:
automated testing

yleeyilin committed:
exceptions handling

yleeyilin committed:
delete with exceptions

yleeyilin committed:
enumerations

yleeyilin committed:
fixed bugs

yleeyilin committed:
update local file access

yleeyilin committed:
revert merge

yleeyilin committed:
level 7 storage

yleeyilin committed:
Merge branch 'branch-Level-7'
to update storage features to allow for caching of list

yleeyilin committed:
added datetime formatting

yleeyilin committed:
Merge branch 'branch-Level-8'
to add in multiple datetime formatting code

yleeyilin committed:
incorporated oops for ui, tasks, storage

yleeyilin committed:
add packages

yleeyilin committed:
Merge remote-tracking branch 'origin/add-gradle-support'
To merge gradle support branch for gradle task

yleeyilin committed:
add junit tests

yleeyilin committed:
refactor for jar

yleeyilin committed:
added javadocs

yleeyilin committed:
abide by coding standard

yleeyilin committed:
added find function

yleeyilin committed:
fixed storage bug

yleeyilin committed:
Merge branch 'branch-A-CodingStandard'
to add javadocs

yleeyilin committed:
merge coding standard

yleeyilin committed:
Merge branch 'branch-Level-9'

yleeyilin committed:
added parser class (missed out earlier)

yleeyilin committed:
Fix naming convention

yleeyilin committed:
Fix checkstyle

yleeyilin committed:
Merge branch 'A-CheckStyle'
Fix and remove the trailing white spaces

yleeyilin committed:
Integrate interface

yleeyilin committed:
Merge branch 'Level-10'
Incorporated interface using javafx technology.

yleeyilin committed:
Fix greeting

MaYuehan commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,31 @@
+package duke;
+
+import java.io.IOException;
+
```
perhaps delete the whitespace between the import to maintain consistency in whitespace usage

MaYuehan commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+    /**
+     * Represents the start time of the event.
+     */
+    protected String start;
```
maybe use startTime to specify the exact meaning of the value it hold?

MaYuehan commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,222 @@
+    public String unmark(String instr, Storage thisStorage) throws DukeException {
+        String res = "";
+        try {
+            int instrNum = Integer.valueOf(instr.split(" ")[1]) - 1;
```
could make the variable name be clearer?

MaYuehan commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,222 @@
+package duke.task;
+
+import java.util.ArrayList;
+
```
perhaps delete the white space?

MaYuehan commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,222 @@
+        case EVENT:
+            try {
+                String eventDescription = instr.substring(6);
+                String[] instrsubString = eventDescription.split(" /from ");
```
maybe change the name to one that can be understand easier?

NusMinato commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,75 @@
+        this.ui = new Ui(System.in);
+        System.out.println(this.ui.greet());
+
+        String instr = this.ui.getInstr();
```
Change the instr to inStr might be better?

NusMinato reviewed (commented):
Hard to find any coding standard issue hahaha. Well done!

yleeyilin committed:
fix exception handling

yleeyilin committed:
Fix exception

yleeyilin committed:
Add assertions
There is a reliance on local file path due to local caching.

Using assertions, we are able to verify the file path:
- "Duke.java": filepath must not be an empty string.
- "Storage.java": filepath must not be an empty string.
- "Parser.java": instructions must not be an empty string.

Using assertions, it ensures that the GUI behaves as expected.

yleeyilin committed:
Merge pull request #1 from yleeyilin/branch-A-Assertions

Add assertions

yleeyilin committed:
Fix code quality
Convoluted code due to deeply nested loops within parser.

- Edit the code to use switch rather than if-else.
- Remove unnecessary nested checks.

Improves readability.

yleeyilin committed:
Merge pull request #2 from yleeyilin/branch-A-CodeQuality

Fix code quality

yleeyilin committed:
Remove unnecessary line

yleeyilin committed:
Fix checkstyle

yleeyilin committed:
Add edit feature
Considered under extensions as C-Update.

Support a way to easily edit details of items without deletion.

yleeyilin committed:
Improve code quality

yleeyilin committed:
Fix header

yleeyilin committed:
Add commit header

yleeyilin committed:
Add system exit feature

Code that reveals what type of exception is not user friendly.

To make it more user friendly, abstract away error messages.

This includes overriding the toString() function.

yleeyilin committed:
Update stage name

yleeyilin committed:
Update README.md

yleeyilin committed:
Update README.md

yleeyilin committed:
Fix format

yleeyilin committed:
Improve code quality

Javadocs had minor violations in the header comments.

This commit has edited the javadocs header to fix these violations.

yleeyilin committed:
Remove extra lines

yleeyilin committed:
Fix exception handling

Interface that does not tell the user the problems with their input is less usable.

To improve usability, I included a customised error message in the UI.

Wrapped exception with duke exception.

yleeyilin committed:
Fix dialog container resizing
